### PR TITLE
Cherry-pick "LibWeb+UI: Miscellaneous key event fixes"

### PR DESCRIPTION
**Pull request by nico**
https://github.com/LadybirdBrowser/ladybird/pull/1699

This renames Key_Alt to Key_LeftAlt (etc), which is a no-op but requires changes to the rest of the system.

As far as I can tell, keymaps and so on still work.

The left/right modifier handling probably needs some future work in Browser to fully work there.

### CODE REVIEW

File: Kernel/API/KeyCode.h [hunk 1]
```diff
@@ -163,7 +165,7 @@ enum KeyModifier {
     Mod_Shift = (1 << 2),
     Mod_Super = (1 << 3),
     Mod_AltGr = (1 << 4),
-    Mod_Keypad = (1 << 5),
+    Mod_Keypad = (1 << 4),
```
Comment (spholz) [inline Kernel/API/KeyCode.h:168]: Keypad should still be `(1 << 5)`, right?

File: Kernel/Devices/HID/VirtIO/Input.cpp [hunk 2]
```diff
@@ -111,7 +111,7 @@ static constexpr auto unshifted_evdev_key_map = to_array<KeyCodeEntry const>({
     { Key_Invalid, 0xff },           { Key_Mute, 0xff },              { Key_VolumeDown, 0xff },        { Key_VolumeUp, 0xff },
     { Key_Power, 0xff },             { Key_Equal, 0xff },             { Key_Invalid, 0xff },           { Key_PauseBreak, 0xff },
     { Key_Invalid, 0xff },           { Key_Invalid, 0xff },           { Key_Invalid, 0xff },           { Key_Invalid, 0xff },
-    { Key_Invalid, 0xff },           { Key_Super, 0xff },             { Key_Super, 0xff },             { Key_Menu, 0xff },
+    { Key_Invalid, 0xff },           { Key_LeftSuper, 0xff },         { Key_LeftSuper, 0xff },         { Key_Menu, 0xff },
```
Comment (spholz) [inline Kernel/Devices/HID/VirtIO/Input.cpp:114]: The right super key should be the second one
https://github.com/SerenityOS/serenity/blob/master/Kernel/Devices/HID/VirtIO/EvDevDefinitions.h#L215

**Review by spholz — COMMENTED** [review]
Just some drive-by comments on the kernel code

I'm a bit confused by what the difference between AltGr and RightAlt is. Shouldn't they be the same key? At least that is how PS/2 and virtio (or rather linux evdev) handle it IIRC.

File: Kernel/Devices/HID/VirtIO/Input.cpp (hunk 2 above)
Comment (nico) [reply Kernel/Devices/HID/VirtIO/Input.cpp:114]: Thanks! This commit intends to just do a behavior-preserving rename. I thought I'd do behavior changes in a follow-up. (I also want to make things that check for left alt or left control check for both alts or both controls.)

File: Kernel/API/KeyCode.h (hunk 1 above)
Comment (nico) [reply Kernel/API/KeyCode.h]: Whoops, yes, I broke this while amending. Fixed.

**Comment by nico** [comment]
> I'm a bit confused by what the difference between AltGr and RightAlt is. Shouldn't they be the same key? At least that is how PS/2 and virtio (or rather linux evdev) handle it IIRC. The HID Usage Tables also only have RightAlt and no AltGr.

Me too! @trflynn89 added AltGr in addition to RightAlt in https://github.com/LadybirdBrowser/ladybird/pull/1699/files#diff-7c30f663f6da38c9cc043a45305e4bbcb347ecb08a22943eed8ded5c6e24c778R26

In some layouts (e.g. `de`), the right alt key acts as altgr and not as alt. E.g. hitting altgr-q produces `@`, while hitting right-alt-q on a layout that doesn't have an altgr key picks "quit" from the file menu (when that's bound to alt-q). Maybe it's supposed to be a post-layout-processing key?

File: Kernel/Devices/HID/VirtIO/Input.cpp (hunk 2 above)
Comment (nico) [reply Kernel/Devices/HID/VirtIO/Input.cpp:114]: (Note to self: When changing this, I'll likely also have to update some key handling code to check for both right and left super then.)

**Comment by spholz** [comment]
[This keycode test website](https://uiwjs.github.io/keycode-info/) says that my altgr key (I have a german keyboard) has a `event.code` of `AltRight` and `event.key` of `AltGraph`. `altKey` is false, probably so it doesn't cause alt keybinds to trigger?
So it sounds like some post processing done by the browser (or maybe ui library?)

Setting my keyboard layout to us doesn't seem to change anything for that key.

File: Kernel/Devices/HID/VirtIO/Input.cpp (hunk 2 above)
Comment (spholz) [reply Kernel/Devices/HID/VirtIO/Input.cpp:114]: (If you want to test this you should set the `SERENITY_EXTRA_QEMU_ARGS` env var to `'-device virtio-tablet -device virtio-keyboard -M i8042=off'`)

**Comment by trflynn89** [comment]
I think when I wrote this, I tested my keyboard on a couple of different mainstream browsers and saw both alt keys have an `event.key` of `Alt`, and `event.code` of `AltLeft`/`AltRight`. Could be remembering that wrong though.

**Pull request merged**